File: 2019/day1.py
# Day 1

def calc_fuel(mass):
  return(int(round(int(mass) / 3) - 2))

# Test
print(calc_fuel(1969))

# Part 1
fuel = 0
with open("day1_input.txt") as f:
  for mod in f:
    fuel += calc_fuel(mod)

print(fuel)

# Part 2
total_fuel = 0

def loop_fuel(fuel):
  other_fuel = 0
  while calc_fuel(fuel) > 0:
    fuel = calc_fuel(fuel)
    other_fuel += fuel
  return other_fuel

with open("day1_input.txt") as f:
  for mod in f:
    total_fuel += loop_fuel(mod)

print(total_fuel)

File: 2019/day1.rb
# Day 1

lines = File.read('Day1_test.txt')
# lines = File.read('Day1_input.txt')

def calc_fuel(mod)
  mod / 3 - 2
end

# Part 1
part1 = 0
lines.each_line { |line|
  part1 += calc_fuel(line.to_i)
}
puts "Part 1 = #{part1}"

# Part 2
part2 = 0

# Test data
lines.each_line { |line|
  fuel = calc_fuel(line.to_i)
  puts "fuel = #{fuel}"
  part2 += fuel
  while fuel > 0 do
    fuel = calc_fuel(fuel.to_i)
    part2 += fuel
  end
}
puts "Part 2 = #{part2}"